src/Xamarin.Android.Build.Tasks/Tasks/BuildApk.cs

```diff
@@ -557,7 +555,7 @@ private void AddGdbservers (ZipArchive apk, ArchiveFileList files, string suppor
 				if (!File.Exists (debugServerPath))
 					continue;
 				Log.LogDebugMessage ("Adding {0} debug server '{1}' to the APK as '{2}'", sabi, debugServerPath, entryName);
-				apk.AddEntry (entryName, File.OpenRead (debugServerPath));
+				apk.AddEntry (File.ReadAllBytes (debugServerPath), entryName);
```

Comment:
Don't we have a Stream-based way to add entries to the .zip? Or are they all `byte[]`-oriented?

If we have streams, we should use them everywhere. If we only have `byte[]`, I shudder at the memory implications as we write a 20+MB `Mono.Android.dll` (unlinked) to the .apk...